File: backend/app/schemas/timesheet.py
from typing import Optional

from pydantic import BaseModel


class TimesheetEntryOut(BaseModel):
    id: int
    consultant_id: int
    consultant_name: Optional[str] = None
    billing_type: Optional[str] = None
    hourly_rate: Optional[float] = None
    entry_date: str
    project_key: Optional[str] = None
    project_name: Optional[str] = None
    task_description: Optional[str] = None
    work_type: Optional[str] = None
    hours: float = 0.0
    notes: Optional[str] = None
    created_at: Optional[str] = None
    updated_at: Optional[str] = None


class TimesheetSummaryOut(BaseModel):
    consultant_id: int
    name: str
    billing_type: Optional[str] = None
    hourly_rate: Optional[float] = None
    project_hours: float = 0.0
    support_hours: float = 0.0
    total_hours: float = 0.0
    tm_cost: float = 0.0

File: frontend/src/types/timesheet.ts
export interface TimesheetEntry {
  id: number;
  consultant_id: number;
  consultant_name: string | null;
  billing_type: string | null;
  hourly_rate: number | null;
  entry_date: string;
  project_key: string | null;
  project_name: string | null;
  task_description: string | null;
  work_type: string | null;
  hours: number;
  notes: string | null;
  created_at: string | null;
  updated_at: string | null;
}

export interface TimesheetSummaryRow {
  consultant_id: number;
  name: string;
  billing_type: string | null;
  hourly_rate: number | null;
  project_hours: number;
  support_hours: number;
  total_hours: number;
  tm_cost: number;
}
